core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/ssp_config.h

```diff
@@ -25,8 +25,8 @@
 
 // On Linux, prefer to use getrandom, though it isn't available in
 // GLIBC before 2.25.
-#if defined(__linux__)                       \
-    && (!defined(__GLIBC__) || __GLIBC__ > 2 \
+#if (defined(__linux__) || defined(ESP_PLATFORM)) \
+    && (!defined(__GLIBC__) || __GLIBC__ > 2      \
         || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 25))
 #define CONFIG_HAS_GETRANDOM 1
```

Comment:
@donghengqaz 
is this glibc version check really relevant to esp-idf?

Reply:
This is a very old modification, I added this to avoid compiling errors.